pkg/cloudproviders/azure/metadata.go

```diff
@@ -79,5 +86,41 @@ func GetMetadata(ctx context.Context) (*storage.ProviderMetadata, error) {
 			},
 		},
 		Verified: verified,
+		Cluster:  clusterMetadata,
 	}, nil
 }
+
+func getClusterMetadata(ctx context.Context) *storage.ClusterMetadata {
+	k8sClient, err := k8sutil.GetK8sInClusterClient()
+	if err != nil {
+		log.Error("Failed to create kubernetes client: ", err)
```

Comment:
typically we would do Errorf with %v for the error, but not sure there is actually any formatting difference